generalhw/hpt630v1/README.md

```diff
@@ -3,21 +3,32 @@
 For details about how this setup works, see
 [../adding-vnc-setup.md](../adding-vnc-setup.md).
 
+### Prepare installation media
+
+Make sure that an OTG drive containing the installation iso is connected to the
+platform using the PiKVM. It can be done using the PiKVM web interface or the
+console according to the instructions [PiKVM docs](https://docs.pikvm.org/msd/).
+
 ### Start the job
 
-Before installation, set booting from installer as a default option (or at least
-after the currently installed OS). As of March 2025, this has to be performed
-manually, as the flow will be different between UEFI and legacy booting. Fix is
-planned to be included when legacy boot is implemented and tested.
+While logged into openQA server (unless you've installed and configured
+`openqa-cli` locally):
 
 Perform installation followed by AEM testing:
 
 ```
 openqa-cli api -X POST isos DISTRI=qubesos VERSION=4.2.3 ARCH=x86_64 BUILD=4.2.3 FLAVOR=install-iso-hpt630v1-uefi
 ```
 
-ISO name is generated as `ISO=Qubes-R%BUILD%-%ARCH%.iso`, you should also create
-a symlink to it in `hdd` directory.  Only Fedora template is going to be
+Use one of these as flavor:
+- `install-iso-hpt630v1-uefi` - UEFI installation, TPM 1.2
+- `install-iso-hpt630v1-legacy` - legacy installation, TPM 1.2
+- `install-iso-hpt630v2-uefi` - UEFI installation, TPM 2.0
+- `install-iso-hpt630v2-legacy` - legacy installation, TPM 2.0
+
+ISO name is generated as `ISO=Qubes-R%BUILD%-%ARCH%.iso`, but it isn't used.
+It is possible to upload the image to PiKVM as part of the test, but it hasn't
+been implemented for this platform yet.  Only Fedora template is going to be
```

Comment:
As a side note: uploading an image to PiKVM is slow at best and at worst results in it rebooting, becoming unresponsive and requires multiple attempts to succeed (depends on PiKVM, network load and phase of the Moon).  So automating can be worse than uploading a suitable image manually.

Also, I suspect iPXE can be made to work there by preparing a fully-functional binary in a format supported by iPXE that comes with the firmware.

Reply:
I've noticed only some of those problems before giving up on the idea.

> Also, I suspect iPXE can be made to work there by preparing a fully-functional binary in a format supported by iPXE that comes with the firmware.

Not all platforms have iPXE.

As something in between, we could create a minimal ISO with iPXE (unless there are some ready-to-use images already) and use that to boot further, but I'm not convinced it will work on all systems. Some UEFI implementations require booting through network in order to expose network protocols that iPXE consumes.

Reply:
I tried booting ISO from https://ipxe.org/download and it works fine on HP (both UEFI and legacy, DHCP takes a long time but succeeds eventually), but on VP4670 it doesn't find any network controllers.

Reply:
VP4670 has some weird network difficulties, that Ethernet-over-USB of PiKVM can be the reason.

> Not all platforms have iPXE.

Sure and using an ISO or a flash/floppy image is also an option (SuperMicro uses a floppy connected through BMC).  It just makes it easier to switch to a different image. Qubes OS 4.2.4 was released a month ago but will need to be uploaded to several PiKVMs for use on all platforms.

It might be fine as is, not sure that we need to try hard at least with the current number of platforms and number of tested versions.

Reply:
VP4670 doesn't work because https://github.com/Dasharo/ipxe/commit/63ed3e352f2e65b26b15a847961440cb4dad0318 isn't upstreamed.

> SuperMicro uses a floppy connected through BMC

IMHO we should stop supporting SuperMicro. AEM won't work there anyway, and its BMC API is much less reliable than PiKVM in my experience.

> Qubes OS 4.2.4 was released a month ago

Yes, and at some point we will have to check if it still works with our tests or not, but we should bring all planned platforms to working state before introducing another unknown.

Reply:
> VP4670 doesn't work because [Dasharo/ipxe@63ed3e3](https://github.com/Dasharo/ipxe/commit/63ed3e352f2e65b26b15a847961440cb4dad0318) isn't upstreamed.

I think it's part of Dasharo+UEFI so upstream status shouldn't be relevant (master branch of that fork seems to be used by Dasharo/coreboot).